.github/workflows/ci.yml

```diff
@@ -21,7 +21,7 @@ jobs:
         uses: ./.github/actions/cache-setup
 
       - name: Install cargo-deadlinks
-        run: cargo install cargo-deadlinks
+        run: which deadlinks || cargo install cargo-deadlinks
```

Comment:
this seems unrelated to the work. Can you provided an updated branch without this commit?

Note that this only runs on CI, and CI has its own logic for caching, installing packages, etc.

Reply:
I updated this line because the CI has [failed](https://github.com/DataDog/glommio/runs/4493424152?check_suite_focus=true) in the first attempt.

```
Run cargo install cargo-deadlinks
    Updating crates.io index
error: binary `cargo-deadlinks` already exists in destination
binary `deadlinks` already exists in destination
Add --force to overwrite
Error: Process completed with exit code 101.
```

Since the error didn't occur in other CIs, it shouldn't happen often, so simply rerunning the CI would have succeeded. At the same time, I didn't feel comfortable ignoring it, so I fixed it in this PR.

I have no objection to removing the commit from this PR and moving it to the other PR.

Reply:
Let's remove it for now, if this gets done, best done in a separate PR. I am bringing this now to @HippoBaro 's attention.

Reply:
I merged for now, because it is indeed failing without it.